File: src/dungeonbreak_narrative/escape_the_dungeon/player/actions.py
"""Player actions and intent modeling."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Any


@dataclass(frozen=True)
class PlayerAction:
    action_type: str
    payload: dict[str, Any]


def action_move(direction: str) -> PlayerAction:
    return PlayerAction(action_type="move", payload={"direction": direction.lower()})


def action_train() -> PlayerAction:
    return PlayerAction(action_type="train", payload={})


def action_talk(target_id: str | None = None) -> PlayerAction:
    payload = {"target_id": target_id} if target_id else {}
    return PlayerAction(action_type="talk", payload=payload)


def action_rest() -> PlayerAction:
    return PlayerAction(action_type="rest", payload={})


def action_search() -> PlayerAction:
    return PlayerAction(action_type="search", payload={})


def action_speak(intent_text: str) -> PlayerAction:
    return PlayerAction(action_type="speak", payload={"intent_text": intent_text})


def action_choose_dialogue(option_id: str) -> PlayerAction:
    return PlayerAction(action_type="choose_dialogue", payload={"option_id": option_id})


def action_live_stream(effort: float = 10.0) -> PlayerAction:
    return PlayerAction(action_type="live_stream", payload={"effort": float(effort)})


def action_steal(target_id: str | None = None) -> PlayerAction:
    payload = {"target_id": target_id} if target_id else {}
    return PlayerAction(action_type="steal", payload=payload)

File: packages/engine/src/escape-the-dungeon/player/actions.ts
import type { PlayerAction } from "../core/types";

export const actionMove = (direction: string): PlayerAction => ({
  actionType: "move",
  payload: { direction: direction.toLowerCase() },
});

export const actionTrain = (): PlayerAction => ({
  actionType: "train",
  payload: {},
});

export const actionTalk = (targetId?: string): PlayerAction => ({
  actionType: "talk",
  payload: targetId ? { targetId } : {},
});

export const actionRest = (): PlayerAction => ({
  actionType: "rest",
  payload: {},
});

export const actionSearch = (): PlayerAction => ({
  actionType: "search",
  payload: {},
});

export const actionSpeak = (intentText: string): PlayerAction => ({
  actionType: "speak",
  payload: { intentText },
});

export const actionFight = (): PlayerAction => ({
  actionType: "fight",
  payload: {},
});

export const actionFlee = (direction: string): PlayerAction => ({
  actionType: "flee",
  payload: { direction: direction.toLowerCase() },
});

export const actionChooseDialogue = (optionId: string): PlayerAction => ({
  actionType: "choose_dialogue",
  payload: { optionId },
});

export const actionLiveStream = (effort = 10): PlayerAction => ({
  actionType: "live_stream",
  payload: { effort },
});

export const actionSteal = (targetId?: string): PlayerAction => ({
  actionType: "steal",
  payload: targetId ? { targetId } : {},
});

export const actionRecruit = (targetId?: string): PlayerAction => ({
  actionType: "recruit",
  payload: targetId ? { targetId } : {},
});

export const actionMurder = (targetId?: string): PlayerAction => ({
  actionType: "murder",
  payload: targetId ? { targetId } : {},
});

export const actionEvolveSkill = (skillId: string): PlayerAction => ({
  actionType: "evolve_skill",
  payload: { skillId },
});

export const actionUseItem = (itemId: string): PlayerAction => ({
  actionType: "use_item",
  payload: { itemId },
});

export const actionEquipItem = (itemId: string): PlayerAction => ({
  actionType: "equip_item",
  payload: { itemId },
});

export const actionDropItem = (itemId: string): PlayerAction => ({
  actionType: "drop_item",
  payload: { itemId },
});
